### DISCUSSION TITLE
Sergej Strelec win10Pe distro

### DISCUSSION THREAD

**Discussion by aigors** [ipxe/ipxe#352] [General]
Hi, there is a usefull win10PE distro made by a Russian Guy https://sergeistrelec.ru/winpe-10-8-sergei-strelec-english/203-winpe-10-8-sergei-strelec-x86-x64-native-x86-20210105-english-version.html

It could be wonderful if will be possibile to boot via Ipxe, any hints to do this? 
it can't booted via memdisk iso due to 6 gbit image.
Many thanks

**Comment by NiKiZe** [comment]
If you have memory, then wimboot could work, otherwise iSCSI.

**Comment by sebaxakerhtc** [comment]
I have it working on my server.
SSTR folder exist in ISO just particularly, full folder is on SMB.
Maybe later (evening) I will share more info.

**Reply by sebaxakerhtc** [reply]
Do you need more info?

**Reply by aigors** [reply]
Of course, should be cool!

**Reply by sebaxakerhtc** [reply]
Ok.
In Legacy BIOS mode iPXE loads sergeistrelec.iso via sanboot
Modded ISO content
![1](https://user-images.githubusercontent.com/32651506/117780486-577aff00-b248-11eb-9eee-5e33dc0e72ad.png)
Also i edit script inside both *.wim like this
![2](https://user-images.githubusercontent.com/32651506/117780493-59dd5900-b248-11eb-9cc6-8ed4671b0fe4.png)
And \\samba\\strelec contents SSTR folder with this content
![3](https://user-images.githubusercontent.com/32651506/117780604-75e0fa80-b248-11eb-931a-fdd3ca98da31.png)

You can also use wimboot to not edit wim images
For EFI mode and behind NAT i use iSCSI
Good luck!

**Reply by aigors** [reply]
Thank you sebaxakerhtc, using this entry in my menù 
`
:strelec 
echo Booting Strelec 
kernel ${nfs-winpe}/wimboot
initrd ${nfs-winpe}/strelec/SSTR/BCD               			BCD
initrd ${nfs-winpe}/strelec/SSTR/boot.sdi          			boot.sdi
initrd ${nfs-winpe}/strelec/SSTR/strelec10Eng.wim        	strelec10Eng.wim`

I get error about BOOTX64.EFI 

Any hints? 
Thanks

**Reply by sebaxakerhtc** [reply]
try to replace boot.sdi with this one for x64
[boot.sdi.zip](https://github.com/ipxe/ipxe/files/6468258/boot.sdi.zip)
or this for x86
[boot.sdi.zip](https://github.com/ipxe/ipxe/files/6468267/boot.sdi.zip)

**Comment by aigors** [comment]
Thank you, but with your boot.sdi i get 
![errore-strelec](https://user-images.githubusercontent.com/2746207/118103961-cee09800-b3da-11eb-8a75-38ecdf97f5c4.png)

**Reply by NiKiZe** [reply]
Normal boot.wim files does contain these files,
either they are missing efi support,
or maybe you are using the x86 and not x64 image on a x64 machine?

**Reply by sebaxakerhtc** [reply]
Looks like you boot in EFI mode.
But for EFI Wimboot you should use `--name`
And you haven't bootmgr.efi 
Add it and try this:

```
:strelec
echo Booting Strelec
kernel ${nfs-winpe}/wimboot
initrd --name bootmgr.efi ${nfs-winpe}/strelec/SSTR/bootmgr.efi bootmgr.efi
initrd --name BCD ${nfs-winpe}/strelec/efi/microsoft/boot/bcd BCD
initrd --name boot.sdi ${nfs-winpe}/strelec/SSTR/boot.sdi boot.sdi
initrd --name boot.wim ${nfs-winpe}/strelec/SSTR/strelec10Eng.wim strelec10Eng.wim
```

**Reply by sebaxakerhtc** [reply]
Tested and work perfect with another bcd and modded network.cmd (added mount of samba share as i show you above)
[bcd.zip](https://github.com/ipxe/ipxe/files/6472077/bcd.zip)

This is my config for EFI x64:
```
kernel test/wimboot || goto failed
initrd --name network.cmd test/network.cmd network.cmd || goto failed
initrd --name bootx64.efi test/efi/boot/bootx64.efi bootx64.efi || goto failed
initrd --name bootmgr test/SSTR/bootmgr bootmgr || goto failed
initrd --name BCD test/bcd BCD || goto failed
initrd --name boot.sdi test/SSTR/boot.sdi boot.sdi || goto failed
initrd --name boot.wim test/SSTR/strelec10x64Eng.wim strelec10x64Eng.wim || goto failed
boot || goto failed
```

**Reply by NiKiZe** [reply]
You should have the same config regardless of efi and pcbios.
You do not need name if the downloaded name is the same, and the error that was given is before that is even relevant.

what you have added in this last post is a reference to bootx64.efi, again that and all the other files, including boot.sdi and BCD is included in standard bootable .wim files, and with the latest wimboot, those files are extracted automatically from wim file.

**Reply by sebaxakerhtc** [reply]
I have simplified config with all files in one folder
![1](https://user-images.githubusercontent.com/32651506/118123349-ca2cdb80-b3fc-11eb-9f0d-c93ef7027353.png)


```
kernel test/wimboot || goto failed
initrd --name network.cmd test/network.cmd network.cmd || goto failed
initrd --name bootx64.efi test/bootx64.efi bootx64.efi || goto failed
initrd --name bootmgr test/bootmgr bootmgr || goto failed
initrd --name BCD test/bcd BCD || goto failed
initrd --name boot.sdi test/boot.sdi boot.sdi || goto failed
initrd --name boot.wim test/strelec10x64Eng.wim strelec10x64Eng.wim || goto failed
boot || goto failed
```

> You should have the same config regardless of efi and pcbios

I can - not should ;) In my instance with strelec i use diferent config files and collection of software for EFI/PC BIOS. 
`UPD 1` And i think *.efi files will not work on PC BIOS platform


> ou do not need name if the downloaded name is the same

Yes, but perfectionist in me will die if i will have row's with --name and without it in one file


> that and all the other files, including boot.sdi and BCD is included in standard bootable .wim files, and with the latest wimboot, those files are extracted automatically from wim file.

Have you tried it with strelec.wim? I like theory, but practice is better way.
`UPD.2` Tested with latest wimboot and no diference. Same error `FATAL: no BOOTX64.EFI or bootmgfw.efi found`


> what you have added in this last post is a reference to bootx64.efi

Because with a standart boot.wim from WinPE you do not need for this, but with strelec do. That's why i say you about practice early

**Reply by sebaxakerhtc** [reply]
Now, after tests i changed network.cmd to remove errors.
```

net use * \\samba\strelec
FOR %%i IN ( c d e f g h i j k l m n o p q r s t u v w y z) DO IF EXIST %%i:\SSTR\WLANProfile SET strelec=%%i:
start /w WLANDRIVER.exe
netcfg.exe -c s -i MS_Server
netcfg.exe -c p -i MS_NDISWAN
netcfg.exe -c p -i MS_NDISWANLEGACY
netcfg.exe -c s -i MS_RASMAN
netcfg.exe -c p -i MS_PPPOE
netcfg.exe -c p -i MS_NDISUIO
netcfg.exe -c s -i ms_nativewifip
netcfg.exe -c s -i ms_vwifi
netcfg.exe -c p -i ms_lltdio
netcfg.exe -c p -i ms_rspndr
netcfg.exe -c p -i ms_lldp
Reg Add "HKLM\SYSTEM\Setup" /v "SystemSetupInProgress" /t REG_DWORD /d "0x0" /f
net start dhcp
net start wcmsvc
net start LanmanServer
net start netprofm
net start MpsSvc
net start Netman
net start nlasvc
net start Server
net start mpssvc
net start RasMan
net start RasAuto
net start wlansvc
net start fdPHost
Reg Add "HKLM\SYSTEM\Setup" /v "SystemSetupInProgress" /t REG_DWORD /d "0x1" /f
wpeutil DisableFirewall
rem wpeutil UpdateBootInfo
FOR /F "tokens=3* delims= " %%i in ('reg query HKLM\HARDWARE\DESCRIPTION\System\BIOS /v BaseBoardManufacturer') DO (SET BaseBoardManufacturer=%%i)
FOR /F "tokens=3* delims= " %%i in ('reg query HKLM\HARDWARE\DESCRIPTION\System\BIOS /v BaseBoardProduct') DO (SET BaseBoardProduct=%%i)
REG add "HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\OEMInformation" /t REG_SZ /v Manufacturer /d "%BaseBoardManufacturer%" /f
REG add "HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\OEMInformation" /t REG_SZ /v Model /d "%BaseBoardProduct%" /f
start /b /wait Reg Add "HKLM\SYSTEM\CurrentControlSet\Control\Windows" /v "ErrorMode" /t REG_DWORD /d "0x00000002" /f
start /b Reg Add "HKLM\SYSTEM\CurrentControlSet\Control\Windows" /v "ErrorMode" /t REG_DWORD /d "0x00000000" /f
cd /D "%strelec%\SSTR\WLANProfile"
for /f "tokens=*" %%i in ('dir /b') do netsh.exe wlan add profile filename="%%i" user=all
exit
```

**Reply by aigors** [reply]
Cool! 
do i need to extract 
network.cmd  
bootx64.efi 
bootmgr 
bcd 
boot.sdi 
From wim file?

**Reply by sebaxakerhtc** [reply]
Only network.cmd from wim. bcd file i attached early
Other files from iso
Look my [comment](https://github.com/ipxe/ipxe/discussions/352#discussioncomment-733282)

**Reply by sebaxakerhtc** [reply]
Nice to be the first in the world who make it working. I saw thousands themes and forums (including iPXE forum) of WWW and nobody and never make it before 🥇 
Later i will publish the instructions on how to make it working

**Reply by aigors** [reply]
Thank you @sebaxakerhtc , in network.cmd i saw net use * \\samba\strelec 
Is the network path of share that contain all iso files from iso images? 
I'm working on synology nas, so, my network samba path  is \\nameofnas\tftproot, in tftprootr i have a folder named strelec with all iso files inside

**Reply by sebaxakerhtc** [reply]
> Thank you @sebaxakerhtc , in network.cmd i saw net use * \samba\strelec
> Is the network path of share that contain all iso files from iso images?
> I'm working on synology nas, so, my network samba path is \nameofnas\tftproot, in tftprootr i have a folder named strelec with all iso files inside

I have just SSTR folder there. For me it's SMB share without wim files.
Just folders I need.
When strelec is loaded it's look for drive with SSTR  folder, that's why I mount strelec folder to drive

**Reply by aigors** [reply]
@sebaxakerhtc 
Now is working for me 
recap 
I have a folder called tftproot that is the root of my netboot server, inside i have a folder named SSTR that is copy of SSTR folder of strelec iso. this folder is shared via samba as \\nameofnas\tftproot
In my network cmd i had to add line  like this 
`net use * \\ipofnas\tftproot /USER:someuser somepassword`
with this line it works like a charm

**Reply by sebaxakerhtc** [reply]
Glad to know it helps.
For Legacy BIOS it will not work. But next week I will publish solution. 
Have a nice weekend! Tomorrow my birthday 🎉🎂

**Reply by F5BJR** [reply]
_or Legacy BIOS it will not work. But next week I will publish solution._

have you a little news ?

**in UEFI mode no problem, vey fine  , many thanks**

![IMG_20210607_145045](https://user-images.githubusercontent.com/32757143/121019756-03f4c600-c7a0-11eb-8e12-40cb93cbe937.jpg)

**Reply by sebaxakerhtc** [reply]
> _or Legacy BIOS it will not work. But next week I will publish solution._
> 
> have you a little news ?

You need to add bootmgr.exe for Legacy boot
![Снимок экрана 2021-06-07 в 17 13 13](https://user-images.githubusercontent.com/32651506/121032117-a66a7480-c7b3-11eb-8a06-7001adcecc22.png)

I attach all necessary files in zip (2,6MB).
https://cloud.burgas.pro/s/strelec/download/strelecipxeboot.zip

You can double-click on Reload Shortcuts on the desktop to see normal Strelec Desktop with icons after a connection to SMB is success

**Reply by F5BJR** [reply]
Many thanks run fine 

problem is with different syntax between UEFI & Legacy !!!

**UEFI good** 

:tftp-strelec
kernel 				SSTR/wimboot || goto failed
initrd --name network.cmd	SSTR/network.cmd network.cmd || goto failed
initrd --name bootx64.efi	SSTR/bootx64.efi bootx64.efi || goto failed
initrd --name **bootmgr**		SSTR/**bootmgr bootmgr** || goto failed
initrd --name BCD		SSTR/BCD BCD || goto failed
initrd --name boot.sdi		SSTR/boot.sdi boot.sdi || goto failed
initrd --name boot.wim		SSTR/strelec10x64Eng.wim **strelec10x64Eng.wim** || goto failed
boot || goto failed

**LEGACY 64** **modified good**  

:tftp-strelec
kernel 				SSTR/wimboot || goto failed
initrd --name network.cmd 	SSTR/network.cmd network.cmd || goto failed
initrd --name bootx64.efi 	SSTR/bootx64.efi bootx64.efi || goto failed
initrd --name bootmgr.exe 	SSTR/**bootmgr64.exe** bootmgr.exe || goto failed
initrd --name BCD 		SSTR/BCD BCD || goto failed 
initrd --name boot.sdi 		SSTR/boot.sdi boot.sdi || goto failed
initrd --name boot.wim 		SSTR/strelec10x64Eng.wim boot.wim || goto failed
boot || goto failed

***
note bootmgr & bootmgr.exe are the same files

**Reply by sebaxakerhtc** [reply]
My config attached. It works like magic for both: Legacy and UEFI platforms. I don't use bootmgr file in my config. This file is from old config in this folder.
Look my iPXE configs

**Reply by NiKiZe** [reply]
EFI --name is used, string after URL is not used.
Pcbios --name not used, string after URL is used.

As for bootmgr, you could add both lines.

**Reply by F5BJR** [reply]
many thanks my friends

I am 75 years old at retirement and I take care of myself a little 

at this time I recycle some material (routers with OpenWrt .

for iPXE at this time i play with a XIAOMI R3G whose HS 48 pins memory replaced by an 8-pins 32 MB SPI 
I use an IDE USB 2.5 box with old 160 Go 

for iPXE running  since this afternoon


All is iPXE for UEFI or Legacy 32 /  64 bits
- Installation for Win7 , 8 , 8.1 , 10 , server 2019 
- Installation for Debian ( netinstall )
- WinPE or Dart 
- slitaz 
- and this day **FULL** strelec  

**thank you again**
